testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveHiddenFiles.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.hive;
+
+import com.google.inject.Inject;
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.assertions.QueryAssert;
+import io.trino.tempto.hadoop.hdfs.HdfsClient;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.util.List;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.tests.product.hive.util.TableLocationUtils.getTablePath;
+import static io.trino.tests.product.utils.QueryExecutors.onHive;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+
+public class TestHiveHiddenFiles
+        extends ProductTest
+{
+    @Inject
+    private HdfsClient hdfsClient;
+
+    @Test
+    public void testSelectFromTableContainingHiddenFiles()
+            throws Exception
+    {
+        String tableName = "test_table_hidden_files" + randomNameSuffix();
+        onTrino().executeQuery("CREATE TABLE " + tableName + " (col integer)");
+
+        onTrino().executeQuery("INSERT INTO " + tableName + " VALUES 1");
+        onTrino().executeQuery("INSERT INTO " + tableName + " VALUES 2");
+        onTrino().executeQuery("INSERT INTO " + tableName + " VALUES 3");
+
+        List<QueryAssert.Row> tableRows = List.of(row(1), row(2), row(3));
+        assertThat(onTrino().executeQuery("SELECT * FROM " + tableName)).containsOnly(tableRows);
+        assertThat(onHive().executeQuery("SELECT * FROM " + tableName)).containsOnly(tableRows);
+
+        String tableLocation = getTablePath(tableName);
+        boolean useDotPrefix = true;
+        // Rename the table files to Hive hidden files (prefixed by `.` or `_` characters)
+        for (String filename : hdfsClient.listDirectory(tableLocation)) {
+            try (ByteArrayOutputStream bos = new ByteArrayOutputStream()) {
+                hdfsClient.loadFile(tableLocation + "/" + filename, bos);
```

Comment:
The renaming of the files could be done much easier with https://github.com/trinodb/tempto/pull/95

cc @ebyhr @findepi